nsqd/nsqd.go

```diff
@@ -203,13 +206,14 @@ func (n *NSQD) getFlag(f int32) bool {
 }
 
 func (n *NSQD) SetHealth(err error) {
-	n.errMtx.Lock()
-	defer n.errMtx.Unlock()
-	n.err = err
 	if err != nil {
 		n.setFlag(flagHealthy, false)
```

Comment:
My preference is for simpler code... but we do have benchmarks, what do they say :grin:?